src/cmds/pbs_rstat.c

```diff
@@ -98,9 +99,14 @@ display_single_reservation(struct batch_status *resv, int how)
 					resv_start = attrp->value;
 				else if (strcmp(attrp->name, ATTR_resv_end) == 0)
 					resv_end = attrp->value;
-				else if (strcmp(attrp->name, ATTR_resv_duration) == 0)
-					resv_duration = atol(attrp->value);
-				else if (strcmp(attrp->name, ATTR_resv_state) == 0)
+				else if (strcmp(attrp->name, ATTR_resv_duration) == 0) {
+					errno = 0;
+					resv_duration = strtol(attrp->value, NULL, 10);
+					if ((errno != 0) || (resv_duration < 0) ) {
+						fprintf(stderr, "Bad Reservation duration. Error: %d\n",errno);
```

Comment:
Now that I think of it, I don't think you should print a message on stderr, it might break tools (like PTL ) which parses pbs_rstat output.

Reply:
Sure @arungrover, we can change this. Can you please suggest, in case of error, how we should handle it? 
I have checked few other code, where in error case, either printing a message in stderr or logging the same. As this tool can be used in client side also, so loggin in server log may not be appropriate. Should I use printf() for error situation ?

Reply:
I'm sorry, I couldn't spend time on this review. I think you are better off not checking the error number of strtol(). Since the value is coming from server, it (PBS server) should have right checks in place to make sure it does not accept bad values from pbs_rsub command. If you think server does not have enough checks in place then we should add that to server.
Since you changed the call from atoi to strtol, it should not give you integer overflow problem anymore.
Sounds reasonable?

Reply:
Sure @arungrover . I will modify the same in code.